### PR TITLE
SPSH-152 Add TRUST_PROXY config for BFF

### PR DESCRIPTION
**Pull request by marode-cap**
# Description
Adds the ability to configure express' "trust proxy" settings for BFF.

See https://expressjs.com/en/guide/behind-proxies.html for a guide.

## Links to Tickets or other pull requests
- https://ticketsystem.dbildungscloud.de/browse/SPSH-152

## Approval for review
- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

### CODE REVIEW

**Review by kristoff-kiefer — COMMENTED** [review]
I'd like @alweber-cap to have a look too because of the decorator

File: src/backend-for-frontend/main.ts [hunk 1]
```diff
@@ -30,6 +31,8 @@ async function bootstrap(): Promise<void> {
     const frontendConfig: FrontendConfig = configService.getOrThrow<FrontendConfig>('FRONTEND');
     const redisConfig: RedisConfig = configService.getOrThrow<RedisConfig>('REDIS');
 
+    app.set('trust proxy', frontendConfig.TRUST_PROXY);
```
Comment (alweber-cap) [inline src/backend-for-frontend/main.ts:34]: This is optional in the config, but needs to be set here. Is setting it to undefined acceptable?

File: src/shared/config/frontend.config.ts [hunk 2]
```diff
@@ -1,11 +1,29 @@
...
 
 export class FrontendConfig {
     @IsInt()
     @Min(1024)
     @Max(10000)
     public readonly PORT!: number;
 
+    @IsOptional()
+    @OneOf(isString, isInt, isBoolean)
+    public readonly TRUST_PROXY?: string | number | boolean;
```
Comment (alweber-cap) [inline src/shared/config/frontend.config.ts:25]: add some example to /config/config.dev.json to understand what is expected here.

File: src/shared/config/frontend.config.ts [hunk 3]
```diff
@@ -1,11 +1,29 @@
...
+import { OneOf } from '../util/one-of.validator.decorator.js';
 
 export class FrontendConfig {
     @IsInt()
     @Min(1024)
     @Max(10000)
     public readonly PORT!: number;
 
+    @IsOptional()
+    @OneOf(isString, isInt, isBoolean)
```
Comment (alweber-cap) [inline src/shared/config/frontend.config.ts:24]: I am unsure if passing predicates or passing types is better here and could not find any best practices so far. We should discuss this @kristoff-kiefer

File: src/backend-for-frontend/main.ts (hunk 1 above)
Comment (marode-cap) [reply src/backend-for-frontend/main.ts:34]: Didn't pose a problem when undefined, but the docs do not clearly state it. Added an undefined-check

File: src/shared/config/frontend.config.ts (hunk 2 above)
Comment (marode-cap) [reply src/shared/config/frontend.config.ts:26]: Set an example value and also linked to the express-docs

**Pull request merged**